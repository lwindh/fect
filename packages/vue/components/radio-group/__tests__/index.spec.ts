import { mount } from '@vue/test-utils'
import Radio from '../../radio'
import RadioGroup from '../index'

describe('Radio', () => {
  it('should be render as a element', () => {
    const wrapper = mount(Radio)
    expect(() => wrapper.unmount()).not.toThrow()
  })
  it('should be support modelVaue to control initial checked', () => {
    const wrapper = mount({
      components: {
        [Radio.name]: Radio,
        [RadioGroup.name]: RadioGroup,
      },
      data() {
        return { val: '1' }
      },
      template: `
      <div class="container">
        <fe-radioGroup v-model="val">
          <fe-radio value="1">Test</fe-radio>
          <fe-radio value="2">Test1</fe-radio>
        </fe-radioGroup> 
      </div>
      `,
    })
    const els = wrapper.findAll('.fect-radio__point')
    expect(els[0].classes('active')).toBe(true)
  })

  it('should be support noraml props like size , disabled , useRow', () => {
    const wrapper = mount({
      components: {
        [Radio.name]: Radio,
        [RadioGroup.name]: RadioGroup,
      },
      data() {
        return { size: 'mini', useRow: true, disabled: true }
      },
      template: `
      <div class="container">
        <fe-radioGroup :size="size" :useRow="useRow" :disabled="disabled">
          <fe-radio value="1">Test</fe-radio>
          <fe-radio value="2">Test1</fe-radio>
        </fe-radioGroup> 
      </div>
      `,
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should be emit update:checked when only use radio and unset disabled', async () => {
    const wrapper = mount(Radio, {
      props: {
        value: '1',
        checked: false,
        disabled: false,
      },
    })
    const el = wrapper.find('input')
    await el.trigger('change')
    expect(wrapper.emitted('update:checked')![0]).toEqual([true])
  })

  it('should emit update:modelValue in radio group', async () => {
    const wrapper = mount({
      components: {
        [Radio.name]: Radio,
        [RadioGroup.name]: RadioGroup,
      },
      data() {
        return { val: '1' }
      },
      template: `
      <div class="container">
        <fe-radioGroup v-model="val">
          <fe-radio value="1">Test</fe-radio>
          <fe-radio value="2">Test1</fe-radio>
        </fe-radioGroup> 
      </div>
      `,
    })
    const el = wrapper.findAll('input')
    const els = wrapper.findAll('.fect-radio__point')
    await el[1].trigger('change')
    expect(els[1].classes('active')).toBe(true)
  })

  it('should be disabled all event when set props disabled', async () => {
    const wrapper = mount(Radio, {
      props: {
        value: 1,
        disabled: true,
      },
    })
    const el = wrapper.find('input')
    await el.trigger('change')
    expect(wrapper.emitted('change')).not.toBeTruthy()
    expect(wrapper.emitted('change')).toBeFalsy()
  })
})
