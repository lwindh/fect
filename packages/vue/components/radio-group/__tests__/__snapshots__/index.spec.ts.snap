// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio should be support noraml props like size , disabled , useRow 1`] = `
<div class="container">
  <div class="fect-radio__group useRow">
    <div class="fect-radio" style="--radioSize: 12px;"><label class="disabled"><input type="radio" disabled=""><span class="fect-radio__name"><span class="fect-radio__point disabled "></span>Test</span></label></div>
    <div class="fect-radio" style="--radioSize: 12px;"><label class="disabled"><input type="radio" disabled=""><span class="fect-radio__name"><span class="fect-radio__point disabled "></span>Test1</span></label></div>
  </div>
</div>
`;
